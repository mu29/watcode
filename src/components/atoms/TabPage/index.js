import { PropTypes } from 'prop-types'
import styled from 'styled-components'

const TabPage = styled.div`
`

TabPage.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.node,
}

export default TabPage
