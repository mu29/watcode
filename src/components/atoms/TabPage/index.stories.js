import React from 'react'
import { storiesOf } from '@storybook/react'
import TabPage from '.'

storiesOf('Atoms/TabPage', module)
  .add('기본', () => (
    <TabPage title="title">
      원대하고, 따뜻한 피어나는 아니더면, 피가 영원히 그들의 이것을 뿐이다. 속에 주며, 것은 가슴에 대고, 것이다.
      두손을 군영과 것은 위하여 넣는 긴지라 위하여서 약동하다. 위하여 동력은 역사를 사막이다.
      풍부하게 있는 우리는 타오르고 주는 산야에 보내는 철환하였는가? 기관과 끓는 그들의 아름다우냐?
      밝은 것은 돋고, 황금시대를 아름다우냐? 같은 못할 듣기만 가치를 하여도 피다. 크고 것은 보이는 어디 석가는 끓는 그리하였는가?
      천하를 그들에게 우리는 보이는 청춘의 끓는다.
    </TabPage>
  ))
