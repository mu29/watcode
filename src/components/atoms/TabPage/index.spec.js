import React from 'react'
import { shallow } from 'enzyme'
import TabPage from '.'

const wrap = (props = {}) => shallow(<TabPage { ...props } />)

describe('<TabPage />', () => {
  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = wrap({ children: 'test', title: 'title' })
    expect(wrapper.contains('test')).toBe(true)
  })

  it('title 속성이 들어온 경우 그려준다', () => {
    const wrapper = wrap({ title: 'foo' })
    expect(wrapper.find({ title: 'foo' })).toHaveLength(1)
  })
})
