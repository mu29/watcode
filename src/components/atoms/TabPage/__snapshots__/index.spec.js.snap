// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPage /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
<div
  className=""
  title="title"
>
  test
</div>
`;

exports[`<TabPage /> 정상적으로 렌더링된다 1`] = `
<div
  className=""
  title="title"
/>
`;
